clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/model/AmazonServerGroup.groovy

```diff
@@ -113,8 +113,20 @@ class AmazonServerGroup implements ServerGroup, Serializable {
       securityGroups = (Set<String>) launchConfig.securityGroups
     }
 
-    if (launchTemplate && launchTemplate.containsKey("securityGroups")) {
-      securityGroups = (Set<String>) launchConfig.securityGroups
+    if (launchTemplate) {
```

Comment:
```
if (launchTemplate) {
  def launchTemplateData = launchTemplate.get("launchTemplateData")
  def securityGroups = (Set<String>) launchTemplateData?.get("securityGroupIds") ?: []

  if (!securityGroups) {
    // attempt to lookup security group identifiers off the first network interface
    def networkInterface = launchTemplateData?.get("networkInterfaces")?.find { it["deviceIndex"] == 0 }
    securityGroups = (Set<String>) networkInterace?.get("groups") ?: []
  }
}
```

I think you might be missing a few null checks.

Reply:
I updated the checks. I had to import `RequestLaunchTemplateData` model in order to move away from `ltData["something"]` syntax, but it still will not let me use the getter methods.